### ISSUE TITLE
Force defineExpose at the end of <script>

### ISSUE THREAD

**Issue by Thomasan1999** [vuejs/eslint-plugin-vue#2235]
**Please describe what the rule should do:**

The rule would force to have `defineExpose` at the end of the `<script>` tag. When `<script setup>` is used, `defineExpose` is used as the alternative to `return` and is often expected to be at the end the same way `return` was. It also uses the data used in `<script>` and should therefore not be placed before the data itself. It is similar to `vue/define-macros-order` where define macros are forced at the beginning of `<script>`.

**What category should the rule belong to?**

[X] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<script setup>
const counter = ref(0);
const title = ref('');

defineExpose({
    counter,
    title,
});

const itemsText = computed(() => `${ counter.value } items`);
</script>

<script setup>
const counter = ref(0);
const title = ref('');

defineExpose({
    counter,
    title,
});

const shown = ref(false);

function show() {
    show.value = true;
}

watch(counter, show);
</script>
```

**Additional context**

**Comment by FloEdelmann** [comment]
I think it could even be part of `vue/define-macros-order` with a new option. What do you think?

**Comment by Eternal-Rise** [comment]
It would be great to have `vue/order-in-components` rule, but for script setup

**Comment by Eternal-Rise** [comment]
@FloEdelmann, I'm not fully agree with that point. If your component have many different logic - probably it is a bad component, that should be decomposed.

**Comment by FloEdelmann** [comment]
@Eternal-Rise True, but please don't discuss that in this issue.

**Comment by J-Michalek** [comment]
> I think it could even be part of `vue/define-macros-order` with a new option. What do you think?

I agree and would welcome `defineExpose` to be added to the `vue/define-macros-order` rule as I ran into issues when I tried to use it right away and it reported invalid rule value.

**Comment by waynzh** [comment]
@FloEdelmann @Thomasan1999 @J-Michalek I can give this a try. 
I recently met this issue as well and put up a [draft PR](https://github.com/vuejs/eslint-plugin-vue/pull/2349).

It differs from the original `order` property, so I added a new Boolean type property called  `defineExposeLast`. However, there are still some implementation details that I may need some help.

- open to suggestions for a better option name…
- Should we include `defineExpose` within the `order` schema? There could potentially have a conflict in the settings, like:
`{ order: ['defineExpose', 'defineProps'], defineExposeLast: true }`
